packages/http/src/validator/index.ts

```diff
@@ -66,12 +59,10 @@ const validateInput: ValidatorFn<IHttpOperation, IHttpRequest> = ({ resource, el
       e => E.right<NonEmptyArray<IPrismDiagnostic>, unknown>(e),
       request =>
         sequenceValidation(
-          request.body ? validateBody(request.body, body, mediaType) : E.right(undefined),
-          request.headers ? headersValidator.validate(element.headers || {}, request.headers) : E.right(undefined),
-          request.query ? queryValidator.validate(element.url.query || {}, request.query) : E.right(undefined),
-          request.path
-            ? pathValidator.validate(getPathParams(element.url.path, resource.path), request.path)
-            : E.right(undefined)
+          request.body ? tryValidateBody(request.body, body, mediaType) : E.right(undefined),
+          request.headers ? validateHeaders(element.headers || {}, request.headers) : E.right(undefined),
+          request.query ? validateQuery(element.url.query || {}, request.query) : E.right(undefined),
+          request.path ? validatePath(getPathParams(element.url.path, resource.path), request.path) : E.right(undefined)
         )
     ),
     E.map(() => element)
```

Comment:
Looks like a candidate to `.bindTo`. Ofc that's totally irrelevant for this PR.

Reply:
Not exactly, bindTo and bind work on the current object returned from the previous chains; here we're mapping to return the original untouched element.

Reply:
Ah, I need to learn a bit about this topic.